.github/workflows/delete-namespace.yml

```diff
@@ -15,22 +15,156 @@ permissions:
-        echo "${BASH_REMATCH[1]}"
-    - name: Set up kubectl
+      - id: create_keycloak_db_name
+        shell: bash
+        run: |
+          # keycloak db name creation at a single point
+          # Change every "." and "-" into "_", then remove all characters that are not alphanumeric, digit or an underscore and cut it on max lenght for postgresql db name
+          temp=$(echo "spsh_keycloak_${{ inputs.namespace }}" | tr - _ | tr \. _ | sed 's/[^a-zA-Z0-9_]//g' | cut -c1-63)
+          echo "before" ${{ inputs.namespace }} "after" $temp
+          echo "keycloak_db_name=$temp" >> $GITHUB_OUTPUT
+
+  create_spsh_app_db_name:
+    runs-on: ubuntu-latest
+    outputs:
+      spsh_app_db_name: ${{ steps.create_spsh_app_db_name.outputs.spsh_app_db_name }}
+    steps:
+      - id: create_spsh_app_db_name
+        shell: bash
+        run: |
+          # spsh-app db name creation at a single point
+          # Change every "." and "-" into "_", then remove all characters that are not alphanumeric, digit or an underscore and cut it on max lenght for postgresql db name
+          temp=$(echo "spsh_database_${{ inputs.namespace }}" | tr - _ | tr \. _ | sed 's/[^a-zA-Z0-9_]//g' | cut -c1-63)
+          echo "before" ${{ inputs.namespace }} "after" $temp
+          echo "spsh_app_db_name=$temp" >> $GITHUB_OUTPUT^
+
+  create_ticket_nr_variable:
+    runs-on: ubuntu-latest
+    outputs:
+      ticket_nr: ${{ steps.extract_branch_ticket_nr.outputs.ticket_nr }}
+    steps:
+      - name: Extract project-ticketnumber from branch
+        id: extract_branch_ticket_nr
+        run: |
+          regex='^([[:alpha:]]+?-[[:digit:]]+)'
+          [[ ${{ inputs.namespace }} =~ $regex ]]
+          echo "ticket_nr=$(echo ${BASH_REMATCH[1]} | tr [A-Z] [a-z])" >> $GITHUB_OUTPUT
+
+  delete_namespace:
+    needs: 
+      - create_keycloak_db_name
+      - create_spsh_app_db_name
+      - create_ticket_nr_variable
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout repo
+      uses: actions/checkout@v2
+      with:
+        repository: 'dBildungsplattform/spsh-app-deploy'
+
+    - name: Set up kubectl and Helm
       uses: azure/setup-kubectl@v3
 
     - name: Configure kubeconfig as tmp file
       run: |
         echo "${{ secrets.SPSH_DEV_KUBECONFIG }}" > kubeconfig
         export KUBECONFIG=$(pwd)/kubeconfig
 
+    - name: Delete all Helm-Charts in Namespace to eliminate all database connections
+      run: | 
+        # Get list of installed Helm-Charts in namespace 
+        LIST_OF_HELM_CHARTS_IN_NAMESPACE="$(helm ls \
+        --namespace ${{ inputs.namespace }} \
+        --kubeconfig $(pwd)/kubeconfig \
+        --short)"
+
+        # Check if list of installed Helm-Charts is empty 
+        if [ -z "$LIST_OF_HELM_CHARTS_IN_NAMESPACE" ]; then
+          echo "No Helm-Charts found in namespace ${{ inputs.namespace }}, skipping uninstall"
+        else
+          helm uninstall \
+            $LIST_OF_HELM_CHARTS_IN_NAMESPACE \
```

Comment:
generally quoting of is advised when expanding bash variables to prevent globbing and word splitting. Especially with a string list of values, i'd be careful

Reply:
simply quoting this like "$LIST_OF_HELM_CHARTS_IN_NAMESPACE" does not work, helm interpret then the whole list as one helm-chart name

Reply:
true, so in this case you do want the spaces preserved. and since the variable is generated in the same action above it should be fine